python/shotgun_desktop/paths.py

```diff
@@ -98,11 +99,19 @@ def get_pipeline_configuration_info(connection):
     )
     logger.debug(pprint.pformat(pcs))
 
-    if len(pcs) == 0:
-        pc = None
-    else:
-        # Pick the last result. See the big comment before the Shotgun query to understand.
-        pc = pcs[-1]
+    pc = None
+    if len(pcs) != 0:
+        # Pick the pipeline configuration with restricted to user, if any.
+        logged_user_id = sgtk.get_authenticated_user().resolve_entity().get("id")
+        for p in pcs:
+            for restricted_user in p.get("users"):
+                if logged_user_id == restricted_user.get("id"):
+                    pc = p
+
+        # If not, pick the last (lowest id).
+        if not pc:
+            pc = pcs[-1]
```

Comment:
I see a few problems here especially when considering the code below (warning if several items are selected).

First, if no configurations (`pcs` items) match our current user, we don't want to select the **last** config of the whole `pcs` list but a subset of `pcs` that only contains configuration with no user restriction.

If we do have a subset of `pcs` with a user restriction that contains `logged_user_id`, we will pick the last one indeed but we need to override `pcs` with this subset to make the code below works.

In other words, you first need to build a subset of `pcs` with the `pc` that contains `logged_user_id`. If not empty, you override `pcs` with it.
Overwise, you need to override `pcs` with a subset of `pc` having no user restrictions.
Then, you pick the last pcs (common to both cases so I believe this part of the code should be unchanged.


Please validate these issues with unit tests first before pushing the fix.